### PR TITLE
Working localization handler

### LINKED ISSUE #21: Check state distributions
Check if all necessary kinds of state distributions are covered and delete unnecessary ones.
Check if values from config are used.
https://github.com/bit-bots/bitbots_navigation/blob/58c98855faf0d1bd8202b542a2ed11126519f237/bitbots_localization/src/StateDistribution.cpp#L7

### LINKED ISSUE #49: Localization: Dynamic reconfiguration does not work
## Expected behavior
Dynamic reconfigure works.

## Current behavior
It doesn't work

## Steps to Reproduce
Try to dynamically reconfigure the localization

### LINKED ISSUE #68: Check localization handler
The localization handler needs to be tested and enhanced.

### PR DESCRIPTION
**Pull request by Flova**
## Proposed changes
* Move localization handler to the right folder
* Add it to the build pipeline and launch file
* Fix multiple bugs in the handler
* Add DSD code to stop the localization during falling->standup

Closes #21 
Closes #49 
Closes #68

### CODE REVIEW

File: bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py [hunk 1]
```diff
@@ -1,87 +1,90 @@
...
     def perform(self, reevaluate=False):
-        rospy.loginfo("initializing position")
+        self.do_not_reevaluate()
+        rospy.logdebug("initializing position")
 
-        rospy.wait_for_service('reset_filter')
-        reset_filter_prox = rospy.ServiceProxy('reset_filter', reset_filter)
+        rospy.wait_for_service('bitbots_localization/reset_filter')
+        reset_filter_prox = rospy.ServiceProxy('bitbots_localization/reset_filter', reset_filter)
+        print(  self.blackboard.poseX,
```
Comment (Flova) [inline bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py:69]: Remove debug print

File: bitbots_localization/src/bitbots_localization/localization_dsd/actions/stop.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import rospy
+from dynamic_stack_decider.abstract_action_element import AbstractActionElement
+from bitbots_localization.srv import set_paused
+
+
+class AbstractLocalizationPause(AbstractActionElement):
+
+    def __init__(self, blackboard, dsd, parameters=None):
+        super(AbstractLocalizationPause, self).__init__(blackboard, dsd, parameters=None)
```
Comment (jaagut) [inline bitbots_localization/src/bitbots_localization/localization_dsd/actions/stop.py:9]: `parameters=None` should be `parameters=parameters`
and below

File: bitbots_localization/src/bitbots_localization/localization_dsd/actions/stop.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        self.stop_filter_prox = rospy.ServiceProxy('bitbots_localization/pause_filter', set_paused)
+
+    def set_paused(self, paused):
+        self.do_not_reevaluate()
+        rospy.wait_for_service('bitbots_localization/pause_filter')
+        try:
+            resp = self.stop_filter_prox(paused)
+        except rospy.ServiceException as e:
+            print(f"Service call failed: {e}")
```
Comment (jaagut) [inline bitbots_localization/src/bitbots_localization/localization_dsd/actions/stop.py:18]: use `rospy.logerr` as above

File: bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py [hunk 4]
```diff
@@ -1,87 +1,90 @@
...
+        rospy.logdebug("initializing pose")
+        rospy.wait_for_service('bitbots_localization/reset_filter')
+        reset_filter_prox = rospy.ServiceProxy('bitbots_localization/reset_filter', reset_filter)
         try:
             resp = reset_filter_prox(0, None, None)
             return resp.success
         except rospy.ServiceException as e:
-            print
-            "Service call failed: %s" % e
+            print(f"Service call failed: {e}")
```
Comment (jaagut) [inline bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py:34]: Replace all these prints with `rospy.logerr`

File: bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py [hunk 5]
```diff
@@ -1,87 +1,90 @@
...
         return
 
 class InitPose(AbstractInitialize):
     def perform(self, reevaluate=False):
-        rospy.loginfo("initializing pose")
-        rospy.wait_for_service('reset_filter')
-        reset_filter_prox = rospy.ServiceProxy('reset_filter', reset_filter)
+        rospy.logdebug("initializing pose")
+        rospy.wait_for_service('bitbots_localization/reset_filter')
+        reset_filter_prox = rospy.ServiceProxy('bitbots_localization/reset_filter', reset_filter)
```
Comment (jaagut) [inline bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py:29]: Please rename all `reset_filter_prox` with `reset_filter_proxy`.

File: bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py (hunk 1 above)
Comment (jaagut) [inline bitbots_localization/src/bitbots_localization/localization_dsd/actions/initialize.py:69]: ?DEBUG?

File: bitbots_localization/src/localization.cpp [hunk 6]
```diff
@@ -458,22 +458,24 @@ void Localization::publish_transforms() {
...
-  estimateMsg.pose.pose.orientation.w = 1;
-  estimateMsg.pose.pose.orientation.x = 0;
-  estimateMsg.pose.pose.orientation.y = 0;
-  estimateMsg.pose.pose.orientation.z = 0;
-  estimateMsg.pose.pose.position.x = 0;
-  estimateMsg.pose.pose.position.y = 0;
+  estimateMsg.pose.pose.orientation.w = q.x();
+  estimateMsg.pose.pose.orientation.x = q.y();
+  estimateMsg.pose.pose.orientation.y = q.z();
+  estimateMsg.pose.pose.orientation.z = q.w();
```
Comment (jaagut) [inline bitbots_localization/src/localization.cpp:471]: Classic...

File: bitbots_localization/src/bitbots_localization/localization_dsd/decisions/decisions.py [hunk 7]
```diff
@@ -27,14 +28,69 @@ class CheckFallen(AbstractDecisionElement):
...
-        if self.blackboard.robot_control_state == 2: #fallen
+        if self.blackboard.robot_control_state == RobotControlState.FALLEN:
             return "FALLEN"
         return "NOT_FALLEN"
 
     def get_reevaluate(self):
         return True
 
 
+class CheckFalling(AbstractDecisionElement):
```
Comment (jaagut) [inline bitbots_localization/src/bitbots_localization/localization_dsd/decisions/decisions.py:39]: Maybe merge these Decisions

**Review by timonegk — CHANGES_REQUESTED** [review]
First set of change requests

**Review by timonegk — APPROVED** [review]
Tested, works fine.

**Pull request merged**